### ISSUE TITLE
[3.5.1] Incorrect generated declaration file

### ISSUE THREAD

**Issue by amanmahajan7** [microsoft/TypeScript#31676]
**TypeScript Version:**  3.5.1, 3.6.0-dev.20190530

**Search Terms:**
declaration, generated types, incorrect default export

**Code**

```
function A() {  }

function B() { }

export function C() {
  return null;
}

C.A = A;
C.B = B;

```

run `tsc --declaration`

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**

```
declare function A(): void;
declare function B(): void;
export declare function C(): null;
export declare namespace C {
    var A: typeof A;
    var B: typeof B;
}
```

**Actual behavior:**

```
declare function A(): void;
declare function B(): void;
declare function C(): null;
declare namespace C {
    var A: typeof A;
    var B: typeof B;
}
export default C;
```

Typescript 3.5.1 is adding a default export which is incorrect.

**Playground Link:** 

**Related Issues:**

**Comment by MayhemYDG** [comment]
It's fine in TS ~3.4 btw.

**Comment by nicojs** [comment]
This breaks a lot of monorepos using project references since they all rely on declaration files. This makes 3.5 unusable for us. Otherwise no problems with the migration though.

**Comment by AnyhowStep** [comment]
A workaround for now,

https://github.com/microsoft/TypeScript/issues/31918#issuecomment-502354686

**Comment by nicojs** [comment]
Cool! Thanks for fixing @weswigham 

I see the 3.6 milestone is linked here. Would it be possible to get this in a patch release instead? I'm pretty desprerate 😅

**Comment by weswigham** [comment]
cc @DanielRosenwasser

**Comment by nicojs** [comment]
Thanks! This issue is now fixed in TS 3.5.3
